### DISCUSSION TITLE
Is it possible to know the provider when using useSession() ?

### DISCUSSION THREAD

**Discussion by jahirfiquitiva** [nextauthjs/next-auth#1537] [Help] [answered]
**Question**
Hello

I was wondering if it's possible to know the provider after the sign in process is completed, and if so how? Because I was checking the session object and it isn't there.

Thanks in advance!

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by balazsorban44** [comment, marked as answer]
not sure about the question here, but I guess you want to save account.provider in the jwt callback and forward it in the session callback?

**Reply by jahirfiquitiva** [reply]
yes, `account.provider` ... but I want it in the `session` object returned by the `useSession` hook, to be used in the client.

**Reply by balazsorban44** [reply]
Then yes, check out the `jwt` and `session` callbacks: https://next-auth.js.org/configuration/callbacks
